web/views.py

```diff
@@ -850,8 +896,14 @@ def user_can_participate(self):
     def get_context_data(self, **kwargs: Any) -> dict[str, Any]:
         context = super().get_context_data(**kwargs)
         response = get_jspsych_response(context)
+        aws_vars = get_jspsych_aws_values()
+        if aws_vars is None:
+            messages.error(
+                self.request,
+                "There was an error starting this study. Please contact lookit@mit.edu.",
+            )
```

Comment:
@okaycj for this error handling, I think I need to make changes to keep the user on the study detail page, rather redirecting to the jsPsych experiment. Do you know whether I need to add a `render` or `redirect` in this case?

Reply:
I think `redirect` is the right approach here but I can't redirect from within `get_context_data`, so my solution was to override the GET method and get the AWS credentials there. That way if there's an error I can still redirect back to the study details (with an error message), otherwise pass the credentials into the context.

Reply:
Overriding GET will get the job done.  You could also try [get_success_url()](https://docs.djangoproject.com/en/5.1/ref/class-based-views/mixins-editing/#django.views.generic.edit.FormMixin.get_success_url).